code/__HELPERS/unsorted.dm

```diff
@@ -1427,6 +1413,46 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 /proc/turf_z_is_planet(turf/T)
 	return GLOB.z_is_planet["[T.z]"]
 
+
+//same as do_mob except for movables and it allows both to drift and doesn't draw progressbar
+/proc/do_atom(atom/movable/user , atom/movable/target, time = 30, uninterruptible = 0,datum/callback/extra_checks = null)
+	if(!user || !target)
+		return 0
+	var/user_loc = user.loc
+
+	var/drifting = 0
+	if(!user.Process_Spacemove(0) && user.inertia_dir)
+		drifting = 1
+
+	var/target_drifting = 0
+	if(!target.Process_Spacemove(0) && target.inertia_dir)
+		target_drifting = 1
+
+	var/target_loc = target.loc
+
+	var/endtime = world.time+time
+//	var/starttime = world.time
+	. = 1
+	while (world.time < endtime)
```

Comment:
timers not this probably

Reply:
this is similar to how the other do_x procs work and uses stoplag to avoid overrunning ticks, timers are not relevant here